File: application/src/test/java/com/smtm/application/security/AuthorizationStepdefs.java
package com.smtm.application.security;

import java.util.List;
import io.cucumber.java.en.Given;

public class AuthorizationStepdefs {

    private final FakeAuthorization fakeAuthorization;

    public AuthorizationStepdefs(FakeAuthorization fakeAuthorization) {
        this.fakeAuthorization = fakeAuthorization;
    }

    @Given("valid tokens are")
    public void validTokensAre(List<String> tokens) {
        fakeAuthorization.addValidTokens(tokens);
    }
}

File: application/src/test/kotlin/com/smtm/application/security/AuthorizationStepdefs.kt
package com.smtm.application.security

import io.cucumber.java.en.Given

class AuthorizationStepdefs(private val fakeAuthorization: FakeAuthorization) {

    @Given("valid tokens are")
    fun validTokensAre(tokens: List<String>) {
        fakeAuthorization.validTokens.addAll(tokens)
    }
}
